File: CSharp/leetcodesln/DecryptStringFromAlphabetToIntegerMapping.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace leetcodesln
{
    public class DecryptStringFromAlphabetToIntegerMapping
    {
        public string FreqAlphabets(string s)
        {
            var ans = new StringBuilder();
            var start = 0;

            while (start < s.Length)
            {
                if(start + 2 < s.Length && s[start+2] == '#')
                {
                    var num = int.Parse(s.Substring(start,2));
                    ans.Append(((char)('a'+num -1)).ToString());
                    start += 3;
                }
                else 
                {
                    ans.Append(((char)('a' + s[start++] - '0'-1)).ToString());
                }
            }
            return ans.ToString();
        }
    }
}

File: Go/questions/DecryptStringFromAlphabetToIntegerMapping.go
package questions

import "strconv"

func FreqAlphabets(s string) string {
	ans := make([]rune, 0)
	start := 0

	for start < len(s) {
		if start+2 < len(s) && s[start+2] == '#' {
			num, _ := strconv.Atoi(s[start : start+2])
			ans = append(ans, rune('a'+uint8(num)-1))
			start += 3
		} else {
			num, _ := strconv.Atoi(s[start : start+1])
			ans = append(ans, rune('a'+uint8(num)-1))
			start++
		}
	}
	return string(ans)
}
